OWML.ModHelper.Input/ModInputHandler.cs

```diff
@@ -0,0 +1,475 @@
+            }
+            if (hash < _maxUsefulKey && _gameBindingCounter[hash] > 0)
+            {
+                return (int)RegistrationCode.CombinationTaken;
+            }
+            return (_comboRegistry.ContainsKey(hash) && !forRemoval ? (int)RegistrationCode.CombinationTaken : hash);
+        }
+
+        public RegistrationCode RegisterCombination(IModInputCombination combination)
+        {
+            var castCombo = (ModInputCombination)combination;
+            castCombo.ClearSingles();
+            if (castCombo == null || castCombo.Combo == null)
+            {
+                _console.WriteLine("combination is null");
+                return RegistrationCode.InvalidCombination;
+            }
+            string[] combs = castCombo.Combo.ToLower().Split('/');
+            List<Int64> combos = new List<Int64>();
+            foreach (string comstr in combs)
+            {
+                Int64 hash = ParseCombination(comstr);
+                if (hash <= 0)
+                {
+                    return (RegistrationCode)hash;
+                }
+                else
+                {
+                    combos.Add(hash);
+                }
+            }
+            foreach (Int64 comb in combos)
+            {
+                _comboRegistry.Add(comb, combination);
+                if (comb < _maxUsefulKey)
+                {
+                    castCombo.AddSingle((KeyCode)comb);
+                }
+            }
+            _logger.Log("succesfully registered " + castCombo.Combo);
+            return RegistrationCode.AllNormal;
+        }
+
+        public RegistrationCode UnregisterCombination(IModInputCombination combination)
+        {
+            var castCombo = (ModInputCombination)combination;
+            if (_comboRegistry.ContainsValue(combination))
+            {
+                if (combination == null || castCombo.Combo == null)
+                {
+                    _console.WriteLine("combination is null");
+                    return RegistrationCode.InvalidCombination;
+                }
+                string[] individualCombos = castCombo.Combo.ToLower().Split('/');
+                List<Int64> hashes = new List<Int64>();
+                foreach (string comboString in individualCombos)
+                {
+                    Int64 hash = ParseCombination(comboString, true);
+                    if (hash <= 0)
+                    {
+                        return (RegistrationCode)hash;
+                    }
+                    else
+                    {
+                        hashes.Add(hash);
+                    }
+                }
+                foreach (Int64 comb in hashes)
+                {
+                    _comboRegistry.Remove(comb);
+                }
+                _logger.Log("succesfully unregistered " + castCombo.Combo);
+                return RegistrationCode.AllNormal;
+            }
+            else
```

Comment:
This else can be removed since you return in the if

Reply:
done